crates/transaction-pool/benches/lookup.rs

```diff
@@ -0,0 +1,63 @@
+#![allow(missing_docs)]
+
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+
+use alloy_primitives::address;
+use reth_transaction_pool::test_utils::{MockTransaction, MockTransactionFactory};
+use std::collections::{BTreeMap, HashMap};
+
+fn bench_map_operations(c: &mut Criterion) {
+    let mut group = c.benchmark_group("map_operations");
+    let mut f = MockTransactionFactory::default();
+
+    // Setup test data
+    let sender = address!("000000000000000000000000000000000000000a");
+    let transactions: Vec<_> = (0..1000)
+        .map(|i| f.validated_arc(MockTransaction::eip1559().with_sender(sender).with_nonce(i)))
+        .collect();
+
+    // Setup maps with same data
+    let btree_data: BTreeMap<_, _> =
+        transactions.iter().map(|tx| (tx.transaction_id, tx.clone())).collect();
+    let hash_data: HashMap<_, _> =
+        transactions.iter().map(|tx| (tx.transaction_id, tx.clone())).collect();
+
+    // Benchmark lookups
+    group.bench_function("btreemap_lookup", |b| {
```

Comment:
These benchmarks can give us a good feel for the actual `Key` and `Value` in play, but they don't tell the full story as `HashMap` is faster than `BTreeMap` for look-up and insertion most of the time. `BTreeMap` still exists for ranged and ordered queries, which are not being benchmarked here. Even if they were, it's hard to speculate which affects the hot paths of an actual mempool more.

I think the most proper way to ensure this is a good change is to write a benchmark for all `pub(crate)` functions of `PendingPool`, or at least the hot ones like `insert_transaction`, `remove_transaction`, `all`, `best`, etc.

Then we can plug in `Vec`, `BTreeMap`, and `HashMap` and see what is best for full nodes, block building nodes, RPC, etc. And what is the best default choice overall? Can parameterize it if there's a clear winner for each different use case.

It's a big ask so maybe we can break this down into smaller issues?

Reply:
I can't really remember most of the stuffs here but let me look at it again